### ISSUE TITLE
note-taking app

### ISSUE THREAD

**Issue by cswarth** [typora/typora-issues#478]
Please, someone who is more skilled then I am, consider making a note-taking app with typora at its center.

Use case for me is taking technical notes including code and images.  Must be able to search notes, and must use markdown.

Evernote - has a really shitty implementation of ` ``` ` codeblocks, but no section headers, no inline code.  Links and images cannot be written as MD.

OneNote - No markdown support at all.  But unlike Evernote, does have section headers.

BoostNote - Really early in the project.  Has markdown support, but preview is in a separate window.  No drag-and-drop image support.  Not clear if this project is still active.

Quiver - Has Markdown support, and drag-and-drom images.  Previews markdown in separate pane.  Project looks abandoned - no further updates for desktop app expected.

**Comment by aquila20** [comment]
This was exactly what I was looking for initially, when I was looking for an alternative to Evernote that runs on MacOS, Linux and Windows. It was nearly impossible to find, and most solutions I did find were buggy or not suited to my needs.

For now I have settled with Typora in combination with an organized folder-system on my local drive. I sync it via Nextcloud, and search my notes via the usual file-search function in the OS. It works very well. 

In short, I understand your request for a note-app, but I do wonder what the added functionality of it would be, given that you can also set it up like I outlined above.

**Comment by cswarth** [comment]
Main added functionality would be a well-organized presentation of notes and notebooks.  And most important, search and good presentation of the results.  Like you, I don't even care about the sync option.  I'll sync plain ascii files via Dropbox.  

I'm not interested in some ground breaking paradigm shift, just a decent implementation of Evernote.   I can't stand that Evernote has a decent platform but a really crappy editor.  They threw in code blocks, but you can only enable them on the version downloaded from the Evernote website, not the one available on the app store.  And then when you do enable the "option" for code blocks, the implementation is  broken in so many ways it is hardly worth the effort.  You would have to really go out of your way to implement code blocks as badly as Evernote managed to do.

**Comment by Tuxii** [comment]
I agree with @cswarth demand. I find Typora editor to be the best markdown editor I ever used. Too bad, it's all it provides. I would love to see some kind of interface to list and search through all my notes, and maybe be able to tag them. 

As Typora seems to be an Electron app with all the logic in JS, I think it wouldn't be too hard to add this feature via a PR if this project was open-sourced.

**Comment by cswarth** [comment]
@Tuxii you might want to checkout the latest version of Boostnote.  While it is not as smooth as typora, they have moved away from the horrible dual pane MD style to a single pane for editing and display.  Their editing is not as smooth as typora as they convert the entire buffer back and forth from MD to display mode instead of just the element under the mouse.

Boostnote has other issues that still make it unusable for daily use, but it seems to be headed in the right direction.

**Comment by Tuxii** [comment]
Thank you @cswarth for letting me know of this app ! This app is really promising !

**Comment by iliyang** [comment]
I just sent an email to hi@typora.io for this feature suggestion. My idea is basically to add a directory browsing section in the left pane, similar to the one in Atom and Sublime Text, that would allow the user to browse and jump between notes within the same window. A search option would of course be the next step. But a file browsing pane would go a long way already!

**Comment by balupton** [comment]
For what it's worth, and as unfortunate as it is in Typora, and open-source and actively maintained, and funded because it is open-source, alternative has arisen: https://github.com/marktext/marktext

So best to build off that project.

Pity that Typora never listened to its users on this one #16

**Comment by abnerlee** [comment]
Closing it since it can be separated into other issues
